tests/zfs-tests/tests/functional/cli_root/zfs_load-key/zfs_load-key_common.kshlib

```diff
@@ -99,3 +99,62 @@ function verify_origin
 
 	return 0
 }
+
+function setup_https
+{
+	log_must openssl req -x509 -newkey rsa:4096 -sha256 -days 1 -nodes -keyout "/$TESTPOOL/snakeoil.key" -out "$SSL_CA_CERT_FILE" -subj "/CN=$HTTPS_HOSTNAME"
+
+	python3 -uc "
+import http.server, ssl, sys, os, time
+
+sys.stdin.close()
+
+httpd, err = None, None
+for i in range(1, 6):
```

Comment:
Is this that fragile? Due to lack of SO_REUSEADDR/SO_REUSEPORT?

Reply:
I was worried my ass has shown, but it seems my assumption was correct: you cannot bind to a port that's already being used for an HTTP connection. I.e. some other process on the machine could be using this port to download something (try this out for yourself with `curl somewhere/big-file | pv -qL 100k > /dev/null`, check what the local address for curl's socket is, and `python3 -m http.server -b localhost that-port-number`).

This simple back-off will (hopefully) let it finish, if that is the case. Could also try other ports here, but that's additional complexity in the users.

Reply:
Actually, randomising the ports was easy enough, so this will now try 100 random ports from the ephemeral range.